src/content_scripts/common.js

```diff
@@ -8,6 +8,9 @@
 let functionalMode = null;
 let fakedColorStatus = null;
 
+// last setting seen by js
+let jsLastColorStatus = null;
```

Comment:
You still need to explain this. Is not this always the same as `fakedColorStatus`? Both are modified (only?) when a new setting is set by the add-on…

Reply:
It is used to decide whether to dispatch events in `dispatchChangeEvents()`.
For example, in case NO_OVERWRITE already means "dark" , and the new setting is "dark".

Reply:
So it is the "original" media query, i.e. the real – not faked one?

If so, just name that variable as such. Though i wonder why you even overwrite it with our value around line 40 here then…